core-bundle/src/Resources/contao/library/Contao/Search.php

```diff
@@ -572,4 +565,38 @@ public static function getInstance()
 
 		return static::$objInstance;
 	}
+
+	/**
+	 * @param string $strUrlA
+	 * @param string $strUrlB
+	 *
+	 * @return int negative if $strUrlA is more canonical, positive if $strUrlB is more canonical
+	 */
+	private static function compareUrls($strUrlA, $strUrlB)
+	{
+		if (strpos($strUrlA, '?') === false && strpos($strUrlB, '?') !== false)
+		{
+			return -1;
+		}
+
+		if (strpos($strUrlA, '?') !== false && strpos($strUrlB, '?') === false)
+		{
+			return 1;
+		}
+
+		$slashCountA = substr_count(explode('?', $strUrlA)[0], '/');
+		$slashCountB = substr_count(explode('?', $strUrlB)[0], '/');
+
+		if ($slashCountA !== $slashCountB)
+		{
+			return $slashCountA - $slashCountB;
+		}
+
+		if (\strlen($strUrlA) !== \strlen($strUrlB))
+		{
+			return \strlen($strUrlA) - \strlen($strUrlB);
+		}
+
+		return strcmp($strUrlA, $strUrlB);
+	}
```

Comment:
It might be more concise to compare with the spaceship operator (+ would also limit the output to {-1, 0, 1} like with the native compare functions).

```suggestion
/**
 * @param string $strUrlA
 * @param string $strUrlB
 *
 * @return int -1 if $strUrlA is more canonical, 1 if $strUrlB is more canonical, 0 if inputs match
 */
private static function compareUrls($strUrlA, $strUrlB)
{
    $containsQueryString = static function(string $str): bool {
        return strpos($str, '?') !== false;
    };
    
    $slashCount = static function (string $str): int {
        return substr_count(explode('?', $str)[0], '/');
    };

    return
        !$containsQueryString($strUrlA) <=> $containsQueryString($strUrlB) ?:
        $slashCount($strUrlA) <=> $slashCount($strUrlB) ?:
        \strlen($strUrlA) <=> \strlen($strUrlB) ?:
        strcmp($strUrlA, $strUrlB);
}
```

Reply:
There is no `<=>` in PHP 5.6

Reply:
:see_no_evil: ... oh man, you're right. I keep forgetting how ancient PHP 5.6 was.